### ISSUE TITLE
Auto-save drafts

### ISSUE THREAD

**Issue by jbasallaje** [verbiricha/habla.news#138]

**Comment by jbasallaje** [comment]
User story

As a user, I want Habla to auto-save my drafts so that if my session is disrupted I won't lose all of my progress.

Related issues:

- https://github.com/verbiricha/habla.news/issues/91
- https://github.com/verbiricha/habla.news/issues/127

Acceptance Criteria:

- Pending

Resources:

- Feedback attached

**Comment by jbasallaje** [comment]
@verbiricha I could see a quick fix version of this not requiring any UI changes. 

I could also see if you wanted to work on this as a part of larger changes to the editor and writer dashboard (issue #53 ), but for now, it's documented here as an issue.

**Comment by Giszmo** [comment]
While remote auto-save would broadcast a user's activity which is a privacy issue, I think it would be appreciated by most users to be able to hop back in where they left off when loggin in from a different device.

I can imagine some extreme cases where maybe some long dormant account would want to announce his come-back in long form and some nerds go crazy about the recent activity ahead of time but for 99% of the users the trade-off would be in favor of just saving it for convenience. If you are very concerned about Fiatjaf some day having his comeback ruined, ask permission on the very first draft saving. If not giving permission, the rejection would have to be stored locally, only of course. A URL parameter `#autoSaveDraft=off` could allow storing this preference in a link.

**Comment by jbasallaje** [comment]
Right, I see what you're saying @verbiricha ; Maybe something like this:

<img width="740" alt="new save modal" src="https://github.com/verbiricha/habla.news/assets/56495918/02d388cb-c5f2-4b7c-9815-5eed60431e5d">

The title and relay selection text are changed and some sub-text at the top of the modal has been added. If that's accurate, I think that's understandable. We can workshop it though.

**Comment by verbiricha** [comment]
Looks great, will update the texts.

**Comment by Giszmo** [comment]
I strongly disagree with the text as it shows a single app mindset where in a protocol you have no control over other clients showing it on the profile or not. See #150

**Comment by verbiricha** [comment]
Perhaps we can clarify that it won't show **in Habla**. You are right that we have no control over what other clients are doing but I doubt that drafts being public is an issue. We can also mitigate this by encrypting the draft contents so other people can't read the work in progress text.

**Comment by Giszmo** [comment]
I would encrypt the full post event in the content of the draft event. No point in leaking mentions and other stuff on an encrypted event. `draft.content=encrypt(JSON.stringify(post))`.

**Comment by jbasallaje** [comment]
I saw this post talking encrypting drafts: nostr:nevent1qqsd0nvsv886mqup3s356upm6clzzsemp98pgh2zcrajt9u790xr0uqpp4mhxue69uhkummn9ekx7mqzyqewrqnkx4zsaweutf739s0cu7et29zrntqs5elw70vlm8zudr3y2qcyqqqqqqgwv6002

I'm not up on it technically, but it seems pertinent. Regarding the text, like you were saying @verbiricha , maybe the subtext could be changed to "Your draft will be saved on nostr, but it will not be visible on your **_Habla_** profile until you publish."
